lib/kitchen/directions/bake_toc.rb

```diff
@@ -59,6 +58,12 @@ def self.li_for_composite_chapter(chapter)
 
       def self.li_for_chapter(chapter)
         pages = chapter.element_children.only(PageElement, CompositePageElement)
+        inner_composite_chapter = chapter.element_children.map do |element|
+          case element
+          when CompositeChapterElement
+            li_for_composite_chapter(element)
+          end
+        end.compact.join("\n")
```

Comment:
To make this follow the existing approach like for `pages`, change this block of added code to

```ruby
inner_composite_chapters = chapter.element_children.only(CompositeChapterElement)
```

and then below line 77 becomes

```ruby
${inner_composite_chapters.map { |composite_chapter| li_for_composite_chapter(composite_chapter) }.join("\n")}
```